adapter/thing_management.md

```diff
@@ -0,0 +1,410 @@
+
+* `interface_spec` is an object with the following structure:
+
+| Field   | Type   | Example            | Description                                                                                                                           |
+|---------|--------|--------------------|---------------------------------------------------------------------------------------------------------------------------------------|
+| int_t   | string | `"out"`            | Type of the interface, either `in` for incoming messages or `out` for outgoing messages.                                              |
+| msg_t   | string | `"evt.pd7.notify"` | Message type or in other words interface name, see [interface format](/fimp/message_format.md#interface-format) for more information. |
+| val_t   | string | `"object"`         | Value type, see [value types ](/fimp/message_format.md#value-types) for more information.                                             |
+| version | string | `"1"`              | Supported version of the protocol.                                                                                                    |
+
+### Examples
+
+* Example of a command to start network inclusion process:
+
+```json
+{
+  "serv": "zwave-ad",
+  "type": "cmd.thing.inclusion",
+  "val_t": "bool",
+  "val": true,
+  "props": {
+    "supports_pin_request": "true"
+  },
+  "tags": null,
+  "src": "-",
+  "ver": "1",
+  "uid": "1e965f4c-07ee-4e3e-8c03-e61e9aa9192a",
+  "topic": "pt:j1/mt:cmd/rt:ad/rn:zw/ad:1"
+}
+```
+
+* Example of a command to stop an ongoing network inclusion process:
+
+```json
+{
+  "serv": "zwave-ad",
+  "type": "cmd.thing.inclusion",
+  "val_t": "bool",
+  "val": false,
+  "props": null,
+  "tags": null,
+  "src": "-",
+  "ver": "1",
+  "uid": "1e965f4c-07ee-4e3e-8c03-e61e9aa9192a",
+  "topic": "pt:j1/mt:cmd/rt:ad/rn:zw/ad:1"
+}
+```
+
+* Example of an event indicating the need of providing a PIN to continue the inclusion process:
+
+```json
+{
+  "serv": "zwave-ad",
+  "type": "evt.thing.pin_needed",
+  "val_t": "string",
+  "val": null,
+  "props": null,
+  "tags": null,
+  "src": "-",
+  "ver": "1",
+  "uid": "1e965f4c-07ee-4e3e-8c03-e61e9aa9192a",
+  "topic": "pt:j1/mt:evt/rt:ad/rn:zw/ad:1"
+}
+```
+
+* Example of a command providing a PIN required to continue the inclusion process:
+
+```json
+{
+  "serv": "zwave-ad",
+  "type": "cmd.thing.pin",
+  "val_t": "string",
+  "val": "12345",
+  "props": null,
+  "tags": null,
+  "src": "-",
+  "ver": "1",
+  "uid": "1e965f4c-07ee-4e3e-8c03-e61e9aa9192a",
+  "topic": "pt:j1/mt:cmd/rt:ad/rn:zw/ad:1"
+}
+```
+
+* Example of an event indicating the current status of the inclusion process:
+
+```json
+{
+  "serv": "zwave-ad",
+  "type": "evt.thing.inclusion_status_report",
```

Comment:
Does FIMP handle including two devices at the same time ? Theoritically might be possible in zigbee-ad.
IF so "evt.thing.inclusion_status_report" would need some device's identifier.

Reply:
I think it should be limited to one at a time, but allowed for multiple serial inclusions during the same session.
So do one from start to end and then start another, but not two at the same time.